From 8547bd612de5b041e5b032823f1570523706e126 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Tue, 9 Nov 2021 15:38:11 +0200
Subject: [PATCH 13/14] s32g3: ddr: Add ECC_OFF configuration mode

Added the DDR code to enable S32G3 platforms to work
without S32G_DRAM_INLINE_ECC=1.

Issue: ALB-8045
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/nxp/s32g/ddr/s32g3/ddrc_cfg.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/drivers/nxp/s32g/ddr/s32g3/ddrc_cfg.c b/drivers/nxp/s32g/ddr/s32g3/ddrc_cfg.c
index ebb242feb..0df21618e 100644
--- a/drivers/nxp/s32g/ddr/s32g3/ddrc_cfg.c
+++ b/drivers/nxp/s32g/ddr/s32g3/ddrc_cfg.c
@@ -46,8 +46,13 @@ struct regconf ddrc_cfg[] = {
 	{0x403c0060, 0x00000000U},
 	{0x403c0064, 0x006100e0U},
 	{0x403c0068, 0x008c0000U},
+#if (S32G_DRAM_INLINE_ECC == 1)
 	{0x403c0070, 0x033f7f54U},
 	{0x403c0074, 0x00000780U},
+#else
+	{0x403c0070, 0x033f7f40U},
+	{0x403c0074, 0x000007b0U},
+#endif
 	{0x403c007c, 0x00000700U},
 	{0x403c00b8, 0x00000000U},
 	{0x403c00bc, 0x00000000U},
@@ -94,13 +99,27 @@ struct regconf ddrc_cfg[] = {
 	{0x403c01b8, 0x00000000U},
 	{0x403c01c0, 0x00000001U},
 	{0x403c01c4, 0x80000001U},
+#if (S32G_DRAM_INLINE_ECC == 1)
 	{0x403c0200, 0x00000014U},
 	{0x403c0204, 0x00050505U},
+#else
+	{0x403c0200, 0x00000017U},
+	{0x403c0204, 0x00080808U},
+#endif
 	{0x403c0208, 0x00000000U},
+#if (S32G_DRAM_INLINE_ECC == 1)
 	{0x403c020c, 0x14141400U},
+#else
+	{0x403c020c, 0x00000000U},
+#endif
 	{0x403c0210, 0x00001f1fU},
+#if (S32G_DRAM_INLINE_ECC == 1)
 	{0x403c0214, 0x04040404U},
 	{0x403c0218, 0x04040404U},
+#else
+	{0x403c0214, 0x07070707U},
+	{0x403c0218, 0x07070707U},
+#endif
 	{0x403c021c, 0x00000f0fU},
 	{0x403c0220, 0x00003f3fU},
 	{0x403c0224, 0x00000000U},
@@ -151,7 +170,11 @@ struct regconf ddrc_cfg[] = {
 	{0x403c0f24, 0x0000ff10U},
 	{0x403c0f2c, 0x00000000U},
 	{0x403c0f38, 0x00000000U},
+#if (S32G_DRAM_INLINE_ECC == 1)
 	{0x403c0f40, 0x37ffffffU},
+#else
+	{0x403c0f40, 0x00000000U},
+#endif
 };
 
 size_t ddrc_cfg_size = ARRAY_SIZE(ddrc_cfg);
-- 
2.17.1

